src/engine/renderer/tr_shader.cpp

```diff
@@ -686,31 +686,29 @@ ParseExpression
 */
 static void ParseExpression( const char **text, expression_t *exp )
 {
-	int            i;
-	char           *token;
-
 	expOperation_t op, op2;
 
 	expOperation_t inFixOps[ MAX_EXPRESSION_OPS ];
-	int            numInFixOps;
+	int numInFixOps = 0;
 
 	// convert stack
 	expOperation_t tmpOps[ MAX_EXPRESSION_OPS ];
-	int            numTmpOps;
-
-	numInFixOps = 0;
-	numTmpOps = 0;
+	int numTmpOps = 0;
 
+	// A ext->numOps equals to 0 means empty or invalid expression.
```

Comment:
Suggesting there is such a thing as a valid empty expression, but I don't suppose that's the case?

Reply:
Maybe an empty expression is invalid, right.

Reply:
I took a closer look and in fact there are "valid" empty expressions, because the parser is buggy, e.g. `((()))`. Apparently they evaluate to 0. It's not something that should have been accepted though.